From bedaf25ab637905e35e87c2f9f7fd686d04e1ef9 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyunk@xilinx.com>
Date: Thu, 19 Sep 2013 19:43:11 -0700
Subject: [PATCH 041/456] drm: xilinx: Add a Xilinx DRM KMS driver to
 Makefile/Kconfig

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Enable and Add a Xilinx DRM KMS driver to Makefile/Kconfig.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit aa76be1358005c42fadb2c7e9ac58a71a141dd7b)

Conflicts:
	drivers/gpu/drm/Kconfig
	drivers/gpu/drm/Makefile
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/Kconfig         |    8 +-------
 drivers/gpu/drm/Makefile        |    5 +----
 drivers/gpu/drm/xilinx/Kconfig  |   11 +++++++++++
 drivers/gpu/drm/xilinx/Makefile |    9 +++++++++
 4 files changed, 22 insertions(+), 11 deletions(-)
 create mode 100644 drivers/gpu/drm/xilinx/Kconfig
 create mode 100644 drivers/gpu/drm/xilinx/Makefile

diff --git a/drivers/gpu/drm/Kconfig b/drivers/gpu/drm/Kconfig
index 8e7fa4d..d1f890d 100644
--- a/drivers/gpu/drm/Kconfig
+++ b/drivers/gpu/drm/Kconfig
@@ -192,10 +192,4 @@ source "drivers/gpu/drm/tilcdc/Kconfig"
 
 source "drivers/gpu/drm/qxl/Kconfig"
 
-source "drivers/gpu/drm/bochs/Kconfig"
-
-source "drivers/gpu/drm/msm/Kconfig"
-
-source "drivers/gpu/drm/tegra/Kconfig"
-
-source "drivers/gpu/drm/panel/Kconfig"
+source "drivers/gpu/drm/xilinx/Kconfig"
diff --git a/drivers/gpu/drm/Makefile b/drivers/gpu/drm/Makefile
index 012cc3f..f18ee64 100644
--- a/drivers/gpu/drm/Makefile
+++ b/drivers/gpu/drm/Makefile
@@ -59,8 +59,5 @@ obj-$(CONFIG_DRM_SHMOBILE) +=shmobile/
 obj-$(CONFIG_DRM_OMAP)	+= omapdrm/
 obj-$(CONFIG_DRM_TILCDC)	+= tilcdc/
 obj-$(CONFIG_DRM_QXL) += qxl/
-obj-$(CONFIG_DRM_BOCHS) += bochs/
-obj-$(CONFIG_DRM_MSM) += msm/
-obj-$(CONFIG_DRM_TEGRA) += tegra/
+obj-$(CONFIG_DRM_XILINX)	+= xilinx/
 obj-y			+= i2c/
-obj-y			+= panel/
diff --git a/drivers/gpu/drm/xilinx/Kconfig b/drivers/gpu/drm/xilinx/Kconfig
new file mode 100644
index 0000000..0e83cfb
--- /dev/null
+++ b/drivers/gpu/drm/xilinx/Kconfig
@@ -0,0 +1,11 @@
+config DRM_XILINX
+	tristate "Xilinx DRM"
+	depends on DRM && ARCH_ZYNQ
+	select DRM_KMS_HELPER
+	select DRM_KMS_CMA_HELPER
+	select DRM_GEM_CMA_HELPER
+	select DRM_ENCODER_ADV7511
+	select SI570
+	select XILINX_AXIVDMA
+	help
+	  DRM display driver for Xilinx IP based pipelines.
diff --git a/drivers/gpu/drm/xilinx/Makefile b/drivers/gpu/drm/xilinx/Makefile
new file mode 100644
index 0000000..a9377fbc
--- /dev/null
+++ b/drivers/gpu/drm/xilinx/Makefile
@@ -0,0 +1,9 @@
+#
+# Makefile for the drm device driver.  This driver provides support for the
+# Direct Rendering Infrastructure (DRI) in XFree86 4.1.0 and higher.
+
+xilinx_drm-y := xilinx_drm_drv.o xilinx_drm_crtc.o xilinx_drm_plane.o \
+	      xilinx_drm_encoder.o xilinx_drm_connector.o
+xilinx_drm-y += xilinx_cresample.o xilinx_rgb2yuv.o xilinx_osd.o xilinx_vtc.o
+
+obj-$(CONFIG_DRM_XILINX) += xilinx_drm.o
-- 
1.7.5.4

